pageserver/src/tenant/timeline.rs

```diff
@@ -1559,14 +1565,17 @@ impl Timeline {
     /// Obtains a temporary lease blocking garbage collection for the given LSN
     pub(crate) fn make_lsn_lease(
         &self,
-        _lsn: Lsn,
+        lsn: Lsn,
         _ctx: &RequestContext,
     ) -> anyhow::Result<LsnLease> {
-        const LEASE_LENGTH: Duration = Duration::from_secs(5 * 60);
         let lease = LsnLease {
-            valid_until: SystemTime::now() + LEASE_LENGTH,
+            valid_until: SystemTime::now() + LsnLease::DEFAULT_LENGTH,
         };
-        // TODO: dummy implementation
+
+        {
+            let mut gc_info = self.gc_info.write().unwrap();
+            gc_info.leases.insert(lsn, lease.clone());
```

Comment:
I feel there is something not optimal of the lease design. Imagine a compute node updates the lease every second and the default lease length is 300s, we will easily accumulate 300 leases in memory. I feel a better design is to have the lease request come with `<compute_unique_id, lsn>`, so that one compute maintains exactly one lease in the system.

Reply:
something to be fixed in the future, could create an issue

Reply:
For this I was thinking that compute will basically consume the `valid_until` expiration time and renew the lease a little bit before that?

Reply:
We keep the leases in a map of LSN->lease, right?  A given compute will keep requesting the same LSN repeatedly, so the overall number of leases we track shouldn't increase.